analyses/malawi/flooding_trigger/mwi-glofas-skill.md

```diff
@@ -0,0 +1,231 @@
+```
+
+### Calculate the skill across lead times
+
+```python
+def is_rainy_season(month):
+    # Oct to April
+    return (month >= 10) | (month <= 4)
+
+def is_dry_season(month):
+    # May to Sept
+    return (month < 10) & (month > 4)
+
+def plot_skill(df_crps, division_key=None, add_line_from_website=False,
+              ylabel="CRPS [m$^3$ s$^{-1}$]"):
+    fig, ax = plt.subplots()
+    for station, ls in zip(STATIONS, [':', '--']):
+        df = df_crps[station].copy()
+        for i, subset in enumerate([None] + skill_thresh_vals[station]):
+            ykey = f'crps_{subset}' if subset is not None else 'crps'
+            y = df[ykey]
+            if division_key is not None:
+                dkey = f'{division_key}_{subset}' if subset is not None else division_key
+                y /= df[dkey]
+            ax.plot(df['leadtime'], y, ls=ls, c=f'C{i}')
+        ax.plot([], [], ls=ls, c='k', label=f'{station}')
+    if add_line_from_website:
+        ax.plot(df_skill[0], df_skill[1], ls='-', c='k', lw=0.5, label='from website')
+    # Add colours to legend
+    for i, subset in enumerate(['full year'] + [f'{thresh}-year return period' for thresh in rp_thresh]):
+        ax.plot([], [], c=f'C{i}', label=subset)
+    ax.set_xlabel("Lead time (days)")
+    ax.set_ylabel(ylabel)
+    ax.legend()
+```
+
+```python
+df_crps = {
+    STATIONS[0]: pd.DataFrame(columns=['leadtime', 'crps']),
+    STATIONS[1]: pd.DataFrame(columns=['leadtime', 'crps'])
+}
+
+for station in STATIONS:
+    for leadtime in da_glofas_reforecast[station].leadtime:
+        forecast = da_glofas_reforecast[station].sel(
+            leadtime=leadtime.values).dropna(dim='time')
+        observations = da_glofas_reanalysis[station].reindex({'time': forecast.time})
+        # For all dates
+        crps = xs.crps_ensemble(observations, forecast,member_dim='number')
+        append_dict = {'leadtime': leadtime.values,
+                                  'crps': crps.values,
+                                   'std': observations.std().values,
+                                   'mean': observations.mean().values,
+                      }
+        # For high values only
+        for thresh in skill_thresh_vals[station]:
+            idx = observations > thresh
+            crps = xs.crps_ensemble(observations[idx], forecast[:, idx], member_dim='number')
+            append_dict.update({
+                f'crps_{thresh}': crps.values,
+                f'std_{thresh}': observations[idx].std().values,
+                f'mean_{thresh}': observations[idx].mean().values
+            })
+        df_crps[station] = df_crps[station].append([append_dict], ignore_index=True)
+```
+
+```python
+plot_skill(df_crps)
+if SAVE_FIG: plt.savefig(PLOT_DIR / 'skill.png')
+plot_skill(df_crps, division_key='mean', ylabel="NCRPS (CRPS / mean)")
+if SAVE_FIG: plt.savefig(PLOT_DIR / 'skill_ncrps.png')
+```
+
+### Investigate bias
+
+```python
+def get_rank(observations, forecast):
```

Comment:
These methods should be made available for all countries in `src`, I am doing this same analysis for Nepal now so I can add it if you'd like